billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/webservice/Client.java

```diff
@@ -0,0 +1,497 @@
+import com.premiumminds.billy.portugal.webservices.documents.DeleteInvoiceRequest;
+import com.premiumminds.billy.portugal.webservices.documents.DocumentTotals;
+import com.premiumminds.billy.portugal.webservices.documents.FatcorewsPort;
+import com.premiumminds.billy.portugal.webservices.documents.FatcorewsPortService;
+import com.premiumminds.billy.portugal.webservices.documents.InvoiceDataType;
+import com.premiumminds.billy.portugal.webservices.documents.InvoiceDataType.LineSummary;
+import com.premiumminds.billy.portugal.webservices.documents.InvoiceStatus;
+import com.premiumminds.billy.portugal.webservices.documents.InvoiceTypeType;
+import com.premiumminds.billy.portugal.webservices.documents.ListInvoiceDocumentsType;
+import com.premiumminds.billy.portugal.webservices.documents.RegisterInvoiceRequest;
+import com.premiumminds.billy.portugal.webservices.documents.ResponseType;
+import com.premiumminds.billy.portugal.webservices.documents.Tax;
+import com.premiumminds.billy.portugal.webservices.documents.TaxType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.xml.datatype.DatatypeConfigurationException;
+import javax.xml.datatype.DatatypeConstants;
+import javax.xml.datatype.DatatypeFactory;
+import javax.xml.datatype.XMLGregorianCalendar;
+import javax.xml.ws.BindingProvider;
+import javax.xml.ws.handler.Handler;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class Client {
+
+    private static final String EFATURA_MD_VERSION = "0.0.1";
+    private static final String AUDIT_FILE_VERSION = "1.04_01";
+    private static final int SUCCESS_CODE = 0;
+
+    private final Config config;
+    private final URL url;
+    private final WebserviceCredentials webserviceCredentials;
+    private final WebserviceKeys webserviceKeys;
+    private final SslClientCertificate sslClientCertificate;
+
+    public Client(Config config, URL url, WebserviceCredentials webserviceCredentials, WebserviceKeys webserviceKeys,
+            SslClientCertificate sslClientCertificate) {
+        this.config = config;
+        this.url = url;
+        this.webserviceCredentials = webserviceCredentials;
+        this.webserviceKeys = webserviceKeys;
+        this.sslClientCertificate = sslClientCertificate;
+    }
+
+    Response sendInvoice(PTApplication application, PTGenericInvoice ptGenericInvoice){
```

Comment:
Any particular reason this method(and the delete methods as well) are package private? Won't this limit their usability?

Reply:
nice catch. thanks
changed